### DISCUSSION TITLE
Best practice? useQuery().promise or suspense?

### DISCUSSION THREAD

**Discussion by borsTiHD** [TanStack/query#8127] [General]
Hi, I would like to "await" for the data in a specific situation.
I found “suspense” and “promise” of “useQuery” in the docs, without being described in detail.
I know that it is still quite new, but for me both do the same job, or what are the differences, or for which use case is which variant intended?

Example for tanstack vue-query:
```ts
const { data, suspense, promise } = useQuery(...)

async function myFunc() {
    await suspense()
    await promise() // or

    // data should be ready
    const myData = data
}
```

**Comment by TkDodo** [comment]
promise is experimental, was added recently and we only made it compatible with the react layer. I don’t know how it would work in vue, if at all. react doesn’t return `suspense` so please stick to that. We should probably omit promise on type level for vue @DamianOsipiuk ?

**Reply by borsTiHD** [reply]
Thank you very much for your response.
This is a great feature, I would be more than happy if it were finally implemented.
Until then, I'll stick with the suggested `suspense`, which is working fine.

**Reply by DamianOsipiuk** [reply]
`suspense` has a [special handling](https://github.com/TanStack/query/blob/main/packages/vue-query/src/useBaseQuery.ts#L145-L186) in vue adapter, so just keep using it.
We might need to override `promise` to be an alias to `suspense` or just remove it from `types` altogether.

There are two features in current `suspense` implementation for `vue` that would not work with `promise`:
- waiting for `reactive` value of `enabled` to become `true` before resolving `suspense`
- resolving `promise` immediately if data in cache is not `stale`
  - this might actually be supported or added to `promise` to immediately resolve with current `data`
